mat64/index_bound_checks.go

```diff
@@ -63,3 +65,47 @@ func (m *Vector) set(r int, v float64) {
 	}
 	m.mat.Data[r*m.mat.Inc] = v
 }
+
+// At returns the element at row r and column c.
+func (t *Symmetric) At(r, c int) float64 {
+	return t.at(r, c)
+}
+func (t *Symmetric) at(r, c int) float64 {
+	if r >= t.mat.N || r < 0 {
+		panic(ErrRowAccess)
+	}
+	if c >= t.mat.N || c < 0 {
+		panic(ErrColAccess)
+	}
+	if t.mat.Uplo == blas.Upper {
+		if r > c {
+			r, c = c, r
+		}
+		return t.mat.Data[r*t.mat.Stride+c]
+	}
+	if r < c {
+		r, c = c, r
+	}
+	return t.mat.Data[r*t.mat.Stride+c]
+}
+
+// Set sets the element at row r and column c. Set panics if the location is outside
+// the appropriate half of the matrix.
```

Comment:
@jonlawlor I agree with you about "wherever possible", but I think this is a case where it's not possible. The documentation for Set is "// Set alters the matrix element at (r, c) to v. It will panic if r or c are out of bounds for the matrix.". For symmetric though, Set alters both the matrix element at (r, c) and the matrix element at (c, r). Seems to me like different behavior, different function name.